test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/ExpressionHelperTest.cs

```diff
@@ -0,0 +1,227 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using Microsoft.AspNetCore.Mvc.ViewFeatures.Internal;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.ViewFeatures
+{
+    public class ExpressionHelperTest
+    {
+        private readonly ExpressionTextCache _expressionTextCache = new ExpressionTextCache();
+
+        public static IEnumerable<object[]> CachedExpressions
+        {
+            get
+            {
+                var key = "TestModel";
+
+                return new TheoryData<Expression>
+                {
+                    (Expression<Func<TestModel, Category>>)(model => model.SelectedCategory),
+                    (Expression<Func<TestModel, CategoryName>>)(model => model.SelectedCategory.CategoryName),
+                    (Expression<Func<TestModel, int>>)(testModel => testModel.SelectedCategory.CategoryId),
+                    (Expression<Func<TestModel, string>>)(model => model.SelectedCategory.CategoryName.MainCategory),
+                    (Expression<Func<TestModel, string>>)(testModel => key),
+                    (Expression<Func<TestModel, TestModel>>)(m => m),
+                };
+            }
+        }
+
+        public static IEnumerable<object[]> IndexerExpressions
+        {
+            get
+            {
+                var i = 3;
+                var key = "TestModel";
```

Comment:
Add a `var List<TestModel> myModel` and confirm expressions using it (e.g. `model => myModel[i].SelectedCategory`) are not cached.